catroid/src/main/java/org/catrobat/catroid/content/bricks/SetBackgroundByIndexBrick.java

```diff
@@ -23,38 +23,47 @@
 
 package org.catrobat.catroid.content.bricks;
 
-import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.EventWrapper;
 import org.catrobat.catroid.content.Sprite;
 import org.catrobat.catroid.content.actions.ScriptSequenceAction;
 import org.catrobat.catroid.formulaeditor.Formula;
 
 import java.util.Collections;
 import java.util.List;
 
-public class SetBackgroundByIndexBrick extends SetLookByIndexBrick {
+import static org.catrobat.catroid.content.EventWrapper.NO_WAIT;
+
+public class SetBackgroundByIndexBrick extends FormulaBrick {
+
+	private static final long serialVersionUID = 1L;
+
+	@EventWrapper.WaitMode
+	protected transient int wait = NO_WAIT;
```

Comment:
this is not necessary. you only use the member `wait` once in the entire class, it is not changed anywhere. just use the value directly:

```@Override
	public List<ScriptSequenceAction> addActionToSequence(Sprite sprite, ScriptSequenceAction sequence) {
		sequence.addAction(sprite.getActionFactory()
				.createSetLookByIndexAction(getSprite(), getFormulaWithBrickField(BrickField.LOOK_INDEX), wait));
				.createSetBackgroundLookByIndexAction(sprite, getFormulaWithBrickField(BrickField.LOOK_INDEX), NO_WAIT));

		return Collections.emptyList();
	}
```

and please have a look if it is used the same way in another brick and remove it there as well. otherwise this is just cargo cult.